docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,101 @@
+- Good, because of its large community support and popularity
+- Good, because it is used by other teams in Mozilla
+- Good, because it no longer uses Redux for state management
+- Bad, because it adds another layer on top of React
+- Bad, because it's new to most team members, and requires some learning to use more complicated features
+- Bad, because it diverges from tooling used elsewhere in the FxA monorepo
+
+### B. React Framework - Gatsby.js
+
+Use a React framework, Gatsby.js. (Gatsby has similar good and bad points to Next.js)
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
+- Good, because of its built in tooling, which reduces existing bespoke tooling currently used
+- Good, because it is easy to learn the basics
+- Good, because it no longer uses Redux for state management
+- Neutral, because of its large community support and popularity, but smaller compared to Next.js
+- Neutral, not currently used by other teams in Mozilla
+- Bad, because it adds another layer on top of React
+- Bad, because it's new to most team members, and requires some learning to use more complicated features
+- Bad, because it diverges from tooling used elsewhere in the FxA monorepo
+
+### C. React, Typescript, without Redux
+
+Use the existing fxa-payments-server app and only remove redux.
+
+- Good, because simplifies the app by removing complexity of using Redux
+- Good, because it would mostly use the existing technologies, and not require new learnings for the team
+- Bad, because Rescripts, CRA, and other tooling make it difficult to upgrade React tooling. (See [FXA-5242](https://mozilla-hub.atlassian.net/browse/FXA-5242))
+- Bad, because it will still require significant refactoring to remove Redux and introduce GraphQL
+- Bad, because page routing with react-router is complicated
+
+### D. React, Typescript, with Redux
+
+Use the existing Subscription Platform, and only refactor existing code.
+
+- Good, because it requires the least amount of new technologies and libraries
+- Bad, because staying with redux keeps complexity in the app without much benefit
+- Bad, because it will still require refactoring to introduce GraphQL, remove application logic, and reduce component complexity
+- Bad, because Rescripts, CRA, and other tooling make it difficult to upgrade React tooling. (See [FXA-5242](https://mozilla-hub.atlassian.net/browse/FXA-5242))
+- Bad, because page routing with react-router is complicated
+
+### E. React Framework - Remix
+
+Use the existing Subscription Platform, and only refactor existing code.
+
+- Good, because it provides an improved developer experience
+- Good, because it provides great page performance
```

Comment:
I don't see page performance mentioned for the other options. What's good about the page performance here? Is it better than the other options in some way?

IIUC app performance isn't a major concern/decision driver, though we will likely see some improvements as a side effect of making fewer network requests and moving more logic/state management to the back end.

Reply:
Time to interact should definitely be considered for any page in a flow we want users to complete as there's pretty decent quantities of studies at this point about falloffs from delay, e.g.
https://fabric.inc/blog/time-to-interact/

Reply:
I explicitly mentioned it here, since the Remix material claims greater performance over competitors like Next. That said, In a discussion during our Engineering Weekly, we concluded that the performance gains are marginal and that Next is nearly as performant.

Reply:
So is this worth mentioning then?

Reply:
I'm actually thinking of doing the opposite and just removing this line. TTI is definitely important, but each of the options considered in this ADR have some means to improve on TTI to bring it within a range that's acceptable.